terraform/firewall_rules.tf

```diff
@@ -1,93 +1,112 @@
-resource "google_compute_firewall" "containerssh_allow_all" {
-  name    = "containerssh-allow-all"
-  network = google_compute_network.main.self_link
+# Firewall rules naming convention:
+# "action_source_to_destination_service/port"
 
-  allow {
-    protocol = "icmp"
+locals {
+  ports = {
+    cadvisor      = "8088"
+    node_exporter = "9100"
+    prometheus    = "9091"
+    minio_server  = "9000"
+    minio_console = "9090"
+    grafana       = "3000"
+    docker_tls    = "2376"
   }
+}
+
+resource "google_compute_firewall" "allow_all_to_network_icmp" {
+  name    = "allow-all-to-network-icmp"
+  network = google_compute_network.main.self_link
   allow {
-    protocol = "udp"
-    ports    = ["0-65535"]
+    protocol = "icmp"
   }
+  source_ranges = ["0.0.0.0/0"]
+}
+
+resource "google_compute_firewall" "allow_all_to_network_ssh" {
+  name    = "allow-all-to-network-ssh"
+  network = google_compute_network.main.self_link
   allow {
     protocol = "tcp"
-    ports    = ["0-65535"]
+    ports    = ["22"]
   }
   source_ranges = ["0.0.0.0/0"]
 }
 
-resource "google_compute_firewall" "containerssh_allow_ssh" {
-  name    = "containerssh-allow-ssh"
-  network = google_compute_network.main.self_link
-
+resource "google_compute_firewall" "allow_all_to_gateway_vm_2333" {
+  name        = "allow-all-to-gateway-vm-2333"
+  description = "Allow access to Gateway VM's SSH server on port 2333"
+  network     = google_compute_network.main.self_link
   allow {
-    protocol = "icmp"
+    protocol = "tcp"
+    ports    = ["2333"]
   }
+  source_ranges = ["0.0.0.0/0"]
+  target_tags   = ["gateway"]
+}
 
+resource "google_compute_firewall" "allow_all_to_logger_vm_grafana" {
+  name    = "allow-all-to-logger-vm-grafana"
+  network = google_compute_network.main.self_link
   allow {
     protocol = "tcp"
-    ports    = ["22"]
+    ports    = [local.ports.grafana]
   }
-
   source_ranges = ["0.0.0.0/0"]
+  target_tags   = ["logger"]
 }
 
-# open port 3000 for Grafana, 9000 and 9090 for MinIO on our logger-vm
-resource "google_compute_firewall" "firewall_logger_view" {
-  name    = "firewall-logger-view"
+resource "google_compute_firewall" "allow_all_to_logger_vm_minio" {
+  name    = "allow-all-to-logger-vm-minio"
   network = google_compute_network.main.self_link
   allow {
     protocol = "tcp"
-    ports    = ["3000", "9000", "9090"]
+    ports = [
+      local.ports.minio_console,
+      local.ports.minio_server
+    ]
   }
-  target_tags   = ["observer"]
   source_ranges = ["0.0.0.0/0"]
+  target_tags   = ["logger"]
 }
 
-# open gateway-port 9100 and 9101, to our prometheus and metrics server
-resource "google_compute_firewall" "firewall_gateway_nodeexport" {
-  name    = "firewall-gateway-nodeexport"
+resource "google_compute_firewall" "allow_all_to_logger_vm_prometheus" {
```

Comment:
still need it for testing metrics server later later.
could close it later

Reply:
ok, i need it as well ^^v